File: 1626-best-team-with-no-conflicts/1626-best-team-with-no-conflicts.java
class Solution {
    public int bestTeamScore(int[] scores, int[] ages) {
        int n=scores.length,ans=Integer.MIN_VALUE;
        int [][]arr=new int[n][2];
        for(int i=0;i<n;i++){
            arr[i][0]=ages[i];
            arr[i][1]=scores[i];
        }
        Arrays.sort(arr,(a,b)->a[0]!=b[0]?a[0]-b[0]:a[1]-b[1]);
        
        int []dp=new int[n];
        for(int i=0;i<n;i++){
            dp[i]=arr[i][1];
            int temp=dp[i];
            for(int j=0;j<i;j++){
                if(arr[i][1]>=arr[j][1]){
                    dp[i]=Math.max(dp[i],dp[j]+temp);
                }
            }
            ans=Math.max(ans,dp[i]);
        }
        return ans;
    }
}

File: 1626-best-team-with-no-conflicts/1626-best-team-with-no-conflicts.cpp
class Solution {
public:
    int bestTeamScore(vector<int>& scores, vector<int>& ages) {
        vector<pair<int,int>>players;
        int n=scores.size();
        int ans=INT_MIN;
        for(int i=0;i<n;i++){
            players.push_back({ages[i],scores[i]});
        }
        sort(players.begin(),players.end(),greater<>());
        vector<int>dp(n);
        for(int i=0;i<n;i++){
            dp[i]=players[i].second;
            int temp=dp[i];
            for(int j=0;j<i;j++){
                if(players[j].second>=players[i].second)
                    dp[i]=max(dp[i],temp+dp[j]);
            }
            ans=max(ans,dp[i]);
        }
        return ans;
    }
}; 